app/controllers/create_survey.rb

```diff
@@ -0,0 +1,53 @@
+get '/surveys/new' do
+  erb :survey_new
+end
+
+
+post '/surveys' do
+  @survey = Survey.new(name: params[:survey_name], user_id: current_user.id)
+    if @survey.save
+      erb :"/question_new", layout: false
+      # redirect "/survey/#{@survey.id}/question/new"
+    else
+      @errors=["You need to put in a survey name"]
+      # erb :'error partial'
+    end
+end
+
+
+get '/surveys/:survey_id/questions/new' do
+  @survey = Survey.find(params[:survey_id])
+  erb :question_new
+end
+
+
+post '/surveys/:survey_id/questions' do
+  @survey= Survey.find(params[:survey_id])
```

Comment:
You need a guard clause here to protect against other people editing this survey. Something like:

```ruby
halt 403 unless @survey.user == current_user
```